chain/network/src/routing/graph_with_cache.rs

```diff
@@ -104,23 +104,24 @@ impl GraphWithCache {
             }
         }
         if removed {
-            *self.cached_next_hops.lock() = None;
+            self.cached_next_hops = OnceCell::new();
         }
     }
 
     /// Computes the next hops table, based on the graph. O(|Graph|).
     pub fn next_hops(&self) -> Arc<NextHopTable> {
-        if let Some(rt) = self.cached_next_hops.lock().clone() {
-            return rt;
-        }
-        let _d = delay_detector::DelayDetector::new(|| "routing table update".into());
-        let _next_hops_recalculation = metrics::ROUTING_TABLE_RECALCULATION_HISTOGRAM.start_timer();
-        trace!(target: "network", "Update routing table.");
-        let rt = Arc::new(self.graph.calculate_distance());
-        metrics::ROUTING_TABLE_RECALCULATIONS.inc();
-        metrics::PEER_REACHABLE.set(rt.len() as i64);
-        *self.cached_next_hops.lock() = Some(rt.clone());
-        rt
+        self.cached_next_hops
+            .get_or_init(|| {
+                let _d = delay_detector::DelayDetector::new(|| "routing table update".into());
+                let _next_hops_recalculation =
+                    metrics::ROUTING_TABLE_RECALCULATION_HISTOGRAM.start_timer();
+                trace!(target: "network", "Update routing table.");
```

Comment:
If we change this `trace` to `span_trace`, we'll get all our three infrastructures to measure perf all in one place!